### PR TITLE
Enable configuration on connection pool on PCF

### LINKED ISSUE #777: Enable DataSourceCloudConfig to configure conneciton pool on PCF
Update dependencies and register as @Bean, also add additional logging to verify connection pool settings are working w/o enabling hikari CP logging.

### PR DESCRIPTION
**Pull request by markpollack**
- Registered DataSourceCloudConfig as a bean
- Added logging
- Added connector deps as done in SCDF, verified final jar has same set of deps

Fixes #777

### CODE REVIEW

File: spring-cloud-skipper-server/pom.xml [hunk 1]
```diff
@@ -45,6 +45,37 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.pivotal.spring.cloud</groupId>
+            <artifactId>spring-cloud-services-starter-config-client</artifactId>
+            <version>1.6.3.RELEASE</version>
```
Comment (jvalkeal) [inline spring-cloud-skipper-server/pom.xml]: Can we define version and excludes only in one place.

Comment (markpollack) [reply spring-cloud-skipper-server/pom.xml]: Let me try again, at the time I ended up playing wack a mole with maven.

**Comment by jvalkeal** [comment]
This PR got reverted.
